### ISSUE TITLE
Style/RedundantInitialize dislikes an empty initialize that changes the arity

### ISSUE THREAD

**Issue by lugray** [rubocop/rubocop#10524]
```ruby
class A
  def initialize(a)
    @a = a
  end
end

class B < A
  def initialize
  end
end

B.new
```

## Expected behavior

All initialize methods accomplish something, so there should be no violation of `Style/RedundantInitialize`.

## Actual behavior

It considers `B#initialize` a violation of `Style/RedundantInitialize`, even though its not redundant. Removing it breaks the final line.

## Steps to reproduce the problem

run `rubocop` with the `Style/RedundantInitialize` cop enabled on the code sample above.

## RuboCop version

```
1.27.0 (using Parser 3.1.1.0, rubocop-ast 1.17.0, running on ruby 2.6.8 arm64-darwin21)
  - rubocop-performance 1.13.3
```

**Comment by koppen** [comment]
I'd like to add another case here.

We've used null-style object for some operations in our codebase. Granted our use case is probably an edge case and warrants a `# rubocop:disable`, but conceptually it looks something like this:

```
class This
  def initialize(stuff)
    @stuff = stuff
  end
end

class This::NullObject
 def initialize(_stuff); end
end
```

`This::NullObject` needs to follow the same API as `This`, removing the initializer is not an option even though it does not change the arity.

**Comment by koic** [comment]
This cop has been reported as false positive and has already been marked as unsafe. Additionally, I've opened #10551 that allows intentional overrides with explicit comment to be allowed by default. So you could suppress the warning by replacing `# rubocop:disable` with an explicit intentional comment.

**Comment by koppen** [comment]
@koic, that's a fine solution. Thank you for your efforts 👍

**Comment by koic** [comment]
@koppen You are welcome!
